OutOfSchool/OutOfSchool.EmailSender/Services/SendGridAccessibilityService.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace OutOfSchool.EmailSender.Services;
+
+public class SendGridAccessibilityService : ISendGridAccessibilityService
+{
+    public SendGridAccessibilityService()
+    {
+        IsSendGridAccessible = true;
+    }
+
+    public bool IsSendGridAccessible { get; set; }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns>Midnight of the next day is SendGrid accessible and null if it is not.</returns>
+    public DateTimeOffset? GetNextStartDate()
+    {
+        if (IsSendGridAccessible)
+        {
+            return null;
+        }
+        return DateTimeOffset.Now.Date.AddDays(1);
+    }
```

Comment:
(Logic) Please help me understand the logic. 

What I see right now is that after first inaccessibility the send grid would never be accessible, because `IsSendGridAccessible` is still never set to `true` after being set to `false`. Am I right?

--- 

My initial suggestion about DateTime was bit different than you have implemented.
I've expected you to have in this class 2 methods and a private field:

```
private DateTimeOffset accessibleAfter = DateTimeOffset.Now;

public void SetSendGridInaccessible(DateTimeOffset now)
  => accessibleAfter = now + ...; // some logic to postpone the next access from now

public bool IsSendGridAccessible(DateTimeOffset now)
  => accessibleAfter <= now;
```

Thus, `job listener` could call `SetSendGridInaccessible(DateTimeOffset.Now)` in case of any issues and `trigger listener` could check `veto` condition using `IsSendGridAccessible(DateTimeOffset.Now)`.

Design remark: `now` is passed in both methods to make `SendGridAccessibilityService` more testable (it's usually hard to test logic that implicitly uses `DateTime.Now`).

And of course the `SendGridAccessibilityService` has to be registered as singleton.